src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,168 @@
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order: " + order, e);
+        }
+        addProducts(order.getId(), order.getProducts());
+        return order;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE deleted = FALSE AND order_id = ?;";
+        Order order = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                order = extractOrder(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't extract orders from DB by ID: "
+                    + id, e);
+        }
+        if (order == null) {
+            return Optional.empty();
+        }
+        order.setProducts(getProducts(order.getId()));
+        return Optional.of(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT * FROM orders WHERE deleted = FALSE";
+        List<Order> orderList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = extractOrder(resultSet);
+                orderList.add(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't extract users from DB", e);
+        }
+        for (Order order : orderList) {
+            order.setProducts(getProducts(order.getId()));
+        }
+        return orderList;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update order: " + order, e);
+        }
+        addProducts(order.getId(), order.getProducts());
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
```

Comment:
I think, we should delete order products from orders_products when we are deleting order.
Same in delete methods in CartDao and UserDao (delete roles in users_roles)

Reply:
that's not corrects, because there will be no point in marking order deleted and removing its relations with products. In that case only remained info will be order and user id and it's not informative